### PR TITLE
docs: Add page to troubleshoot multiple versions of a package

### PR DESCRIPTION
**Pull request by manueliglesias**
*Issue #, if available:*
N/A

*Description of changes:*
Added a page with upgrade instructions and troubleshooting

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/troubleshooting/fragments/js/upgrading.md
Comment (medelman17) [inline docs/lib/troubleshooting/fragments/js/upgrading.md:1]: Can you include the likely error-scenarios that would lead one to this documentation? For example, "If you're seeing... X, Y, Z, then you may want to check the following..."?

File: docs/lib/troubleshooting/fragments/js/upgrading.md
Comment (medelman17) [inline docs/lib/troubleshooting/fragments/js/upgrading.md:60]: I think there may be a typo in the above sentence--not sure I grasp the meaning as written.

File: docs/lib/troubleshooting/fragments/js/upgrading.md
Comment (medelman17) [inline docs/lib/troubleshooting/fragments/js/upgrading.md:154]: Can we make it more clear what to do in the non-happy-path situation?

**Review by medelman17 — COMMENTED** [review]
Great start! Thanks for this!

File: docs/lib/troubleshooting/fragments/js/upgrading.md
Comment (manueliglesias) [reply docs/lib/troubleshooting/fragments/js/upgrading.md:154]: Thanks Mike, the command outputs the latest versions of amplify packages **published to npm**, it doesn't look at your local dependencies. It is just a convenient way to see the `@latest` versions without having to go to the repo's homepage, look at the commits, search for the one for the release and look into the commit message.

The command is useful mostly for maintainers when helping customers. Maybe I should just remove it?

File: docs/lib/troubleshooting/fragments/js/upgrading.md
Comment (manueliglesias) [reply docs/lib/troubleshooting/fragments/js/upgrading.md:60]: Oops, thanks, fixed

File: docs/lib/troubleshooting/fragments/js/upgrading.md
Comment (manueliglesias) [reply docs/lib/troubleshooting/fragments/js/upgrading.md:1]: Once aws-amplify/amplify-js#8083 is merged, customers most likely will arrive here coming from the [bug issue template checklist](https://github.com/aws-amplify/amplify-js/pull/8083/files#diff-7e6dd26cf81256b658eb44be8a5664393c3d886d8de97606d7f89f691d98b48d) or the [messages logged by the library](https://github.com/aws-amplify/amplify-js/pull/8083/files#diff-db69e785d7fc5c73340d7a6b0eb17b7804601ee6143bb4d95df2968ace533c80)

Would you like to see the scenarios being mentioned here too?

Comment (medelman17) [reply docs/lib/troubleshooting/fragments/js/upgrading.md:1]: I would. We can't anticipate how customers will arrive at a given documentation resource. Theoretically, folks could arrive via search. Also, we're including the resource in our top-level navigation. In other words, "Every page is page one." We should provide more context, here.

File: docs/lib/troubleshooting/fragments/js/upgrading.md
Comment (medelman17) [reply docs/lib/troubleshooting/fragments/js/upgrading.md]: If it's mostly useful for maintainers, I would remove it. I empathize with the fact we lack a robust, internal help system / knowledge base.

**Review by medelman17 — APPROVED** [review]

**Pull request merged**
